### PR TITLE
Bug 1527428 - fix removal of listeners, and add tests

### PR DESCRIPTION
**Pull request by djmitche**
Bugzilla Bug: [1527428](https://bugzilla.mozilla.org/show_bug.cgi?id=1527428)

### CODE REVIEW

File: services/hooks/src/listeners.js [hunk 1]
```diff
@@ -76,31 +80,29 @@ class HookListeners {
...
+    this.listeners[queueName] = listener;
   }
 
   /** Delete a listener for the given queueName  */
   async removeListener(queueName) {
     debug(`${queueName}: stop listening`);
-    let removeIndex = this.listeners.findIndex(({_queueName}) => _queueName === queueName);
-    if (removeIndex > -1) {
-      const listener = this.listeners[removeIndex];
+    const listener = this.listeners[queueName];
```
Comment (djmitche) [inline services/hooks/src/listeners.js]: This is the key bit -- the `_queueName` search was not working (and after all is depending on an internal thing).  The rest is just tests and some minor refactorings.

File: services/hooks/src/listeners.js [hunk 2]
```diff
@@ -57,15 +57,19 @@ class HookListeners {
...
-    this.listeners = [];
+    this.listeners = {};
     // Reconcile on start up
     await this.reconcileConsumers();
   }
 
   /** Create a new pulse consumer for a hook */
-  async createListener(hook, queueName) {
+  async createListener(hookGroupId, hookId, queueName) {
     debug(`${queueName}: creating listener (and queue if necessary)`);
```
Comment (imbstack) [inline services/hooks/src/listeners.js]: Should this come after the new check below?

File: services/hooks/src/listeners.js [hunk 3]
```diff
@@ -76,31 +80,29 @@ class HookListeners {
...
+    if (listener) {
       await listener.stop();
-      this.listeners.splice(removeIndex, 1);
     }
   }
 
   haveListener(queueName) {
-    const index = this.listeners.findIndex(({_queueName}) => _queueName === queueName);
-    return index > -1;
+    return !!this.listeners[queueName];
```
Comment (owlishDeveloper) [inline services/hooks/src/listeners.js]: Isn't `if (!this.haveListener(queue.queueName))` the same as `if (!this.listeners[queueName])`?

File: services/hooks/src/listeners.js (hunk 2 above)
Comment (djmitche) [reply services/hooks/src/listeners.js]: Probably!

File: services/hooks/src/listeners.js (hunk 3 above)
Comment (djmitche) [reply services/hooks/src/listeners.js]: Yeah, that might be simpler.

**Pull request merged**